bids/analysis/base.py

```diff
@@ -45,13 +52,13 @@ def setup(self):
 
 class Block(object):
 
-    def __init__(self, analysis, level, index, name, transformations=None,
+    def __init__(self, analysis, level, index, name=None, transformations=None,
                  model=None, contrasts=None):
 
         self.analysis = analysis
         self.level = level
         self.index = index
-        self.name = name
+        self.name = name or level
```

Comment:
This won't work because `level` is not guaranteed to be unique, and we need to be able to uniquely identify Blocks. If we make `name` optional, we at minimum need to inspect all levels when initializing `Analysis` and raise an error if there are redundant values without names.

If we keep `name` optional, we still shouldn't assign `level` to `self.name` because that will create all kinds of ambiguity (e..g, the user now won't have a clear way of knowing whether `analysis['run']` is going to work or not, and if so, what it's going to return.